internal/signer/linux/pkcs11/pkcs11.go

```diff
@@ -183,6 +183,9 @@ func (k *Key) encryptRSA(hash hash.Hash, data []byte) ([]byte, error) {
 func (k *Key) encryptRSAWithPKCS11(data []byte) ([]byte, error) {
 	publicKeyFilter := pkcs11.Filter{pkcs11.ClassPublicKey, ""}
 	pubObjs, err := (k.slot).Objects(publicKeyFilter)
+	if pubObjs == nil {
+		return nil, fmt.Errorf("encryptRSAWithPKCS11 error: no public keys found")
+	}
```

Comment:
I think this is redundant with the `err != nil` check on line 190, if you just move the array access after err.

Reply:
Maybe also add an `or` statement that checks pubObjs has a length greater than 0, after you check nil

Reply:
Okay, I updated the function to check the err first and then the length of pubObjs.